test/cpp/qps/client_async.cc

```diff
@@ -236,33 +236,46 @@ class AsyncClient : public ClientImpl<StubType, RequestType> {
     this->EndThreads();  // this needed for resolution
   }
 
-  void InitThreadFunc(size_t thread_idx) override final {}
-  bool ThreadFunc(HistogramEntry* entry, size_t thread_idx) override final {
+  void ThreadFunc(size_t thread_idx, Client::Thread* t) override final {
     void* got_tag;
     bool ok;
 
-    if (cli_cqs_[cq_[thread_idx]]->Next(&got_tag, &ok)) {
+    HistogramEntry entry;
+    HistogramEntry* entry_ptr = &entry;
+    if (!cli_cqs_[cq_[thread_idx]]->Next(&got_tag, &ok)) {
+      return;
+    }
+    ClientRpcContext* ctx = nullptr;
+    std::mutex* shutdown_mu = &shutdown_state_[thread_idx]->mutex;
+    auto cq_lambda = [&, ctx, ok, entry_ptr, shutdown_mu]() {
```

Comment:
F&& change should allow this to be passed to DoThen directly, which would be preferable